Framework/Algorithms/test/WorkspaceGroupTest.h

```diff
@@ -296,6 +296,39 @@ class WorkspaceGroupTest : public CxxTest::TestSuite {
     AnalysisDataService::Instance().remove("InputWS");
   }
 
+  void testRenameWorkspaceDuplicates() {
```

Comment:
This is a great test to add but can we also add a separate test for renaming of a workspace that is part of a group to a name that doesn't already exist in the group? That just covers the other use case here then.